### PR TITLE
274 - Update for AutoGrow and MaxLength Option in TextArea

### PR DESCRIPTION
**Pull request by tjamesallen15**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

Added configuration for autoGrow and maxLength settings.

**Related github/jira issue (required)**:

Partial Work for #274 

**Steps necessary to review your pull request (required)**:

Build the project and go to textarea-demo. Check the max chars and two growable textarea.

### CODE REVIEW

**Comment by tjamesallen15** [comment]
Hi Guys,

I created a new PR and closed the old ones because I need to fix where I created the branches. This is it. @bthharper  @pwpatton @tmcconechy 

Though I'm not much sure about what is the error right now in the build.

Thanks.

**Comment by tjamesallen15** [comment]
@tmcconechy 
I also tried npm run lint before pushing and creating a PR and it doesn't have an error. However when I created the PR, the travis build for lint test contains error.

**Comment by tmcconechy** [comment]
Is this ok for you know @pwpatton ?

**Comment by pwpatton** [comment]
I'm concerned that skipping the inputs and going right to a settings object, while it might work here, is bad practice since our markForCheck call logic (and hence updating of the components view) is done in the input setter methods.

I'd like to see the changes like this: 
`this.textStatic.settings.autoGrowMaxHeight = autoGrowMaxHeight;`

Changed to this
`this.textStatic.autoGrowMaxHeight = autoGrowMaxHeight;`

**Comment by tmcconechy** [comment]
Ok @tjamesallen15 can you have a look and address @pwpatton comments? Thanks!

**Comment by tjamesallen15** [comment]
Hi @tmcconechy and @pwpatton,

I just want to ask some clarification since I'm just following the code order in every component.

@Input() set autoGrowMaxHeight(autoGrowMaxHeight: number) {
    this.options.autoGrowMaxHeight = autoGrowMaxHeight;
    if (this.textStatic) {
      this.textStatic.settings.autoGrowMaxHeight = autoGrowMaxHeight;
    }
  }

In this snippet, the options comes first before the settings. Am I missing something? I'm just confuse. Thank you!

File: projects/ids-enterprise-ng/src/lib/textarea/soho-textarea.component.ts [hunk 1]
```diff
@@ -73,8 +74,38 @@ export class SohoTextAreaComponent extends BaseControlValueAccessor<string> impl
   @HostBinding('class.resizable')
   @Input() resizable: boolean = null;
 
-  @HostBinding('attr.maxlength')
-  @Input() maxlength: number;
+  @Input() set maxLength(maxLength: number) {
+    this.options.maxLength = maxLength;
+    if (this.textStatic) {
+      this.textStatic.settings.maxLength = maxLength;
```
[the remark this answers is not in the document]
Comment (pwpatton) [reply projects/ids-enterprise-ng/src/lib/textarea/soho-textarea.component.ts]: My bad this look fine

**Pull request merged**